components/connectionmanager/src/lib.rs

```diff
@@ -203,4 +213,33 @@ impl ConnectionManager {
         }
         join_all(futures).await;
     }
+
+    fn dns_seed(self: &Arc<Self>, mut min_addresses_to_fetch: usize) {
+        let mut dns_seeders = self.dns_seeders.to_vec();
+        dns_seeders.shuffle(&mut thread_rng());
```

Comment:
FYI, You can use https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html#tymethod.choose_multiple to do this without copying the seeders to a Vec.